apps/tests/devapps/client_secret_sample.go

```diff
@@ -11,23 +11,28 @@ import (
 	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
 )
 
+// Keep the ConfidentialClient application object around, because it maintains a token cache
+var _app *confidential.Client
```

Comment:
Shouldn't the other `client_certificate_sample.go` file also pick up such a change?

Reply:
Done.